Update release documentation (#229)
kyma-project/community: guidelines/releases/release-process.md

Prow triggers the jobs again. Return to point 6 to rerun manual jobs.

11. After all checks pass, merge the PR.

> >**NOTE:** Bear in mind that merging to the release branch requires approvals from all teams.

Reviewer: ```suggestion
>**NOTE:** To merge the PR to the release branch, you must receive approvals from all teams.
```

12. Merging the PR to the release branch runs the postsubmit job that creates a GitHub release.
Validate the `yaml` and changelog files generated under [releases](https://github.com/kyma-project/kyma/releases).
Update the release content manually with the instruction on how to install the latest Kyma release.

13. Update `RELEASE_VERSION` to the next version both on the `master` and release branches. Do it immediately after the release, otherwise any PR to a release branch overrides the previously published Docker images. 
 

> >**NOTE:** Remember that Release Candidate versions should be tested by all teams. To make it easier, provision a publicly available cluster with the RC version after performing all the steps listed above.

Reviewer: ```suggestion
>**NOTE:** All teams should test the release candidate versions. To make the testing easier, provision a publicly available cluster with the release candidate version after performing all steps listed in this document.
```
